### PR TITLE
Relations example fix

### PR DESCRIPTION
**Pull request by mhwelander**

### CODE REVIEW

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx [hunk 1]
```diff
@@ -6,28 +6,35 @@ metaDescription: 'A relation is a connection between two models in the Prisma sc
...
 }
 ```
 
-> **Note**: In SQL, relations between models are represented via _foreign keys_ in the underlying database. A foreign key is only stored on _one_ side of a relation. As opposed to how relations are modeled in SQL, **Prisma always requires explicit relation fields on _both_ sides of the relation to be set on your Prisma models**.
->
-> This means that one side of the relation represents the _foreign key_ in the underlying database, the other relation field is provided on a Prisma-level (in the case of implicit many-to-many relations, both relation fields are provided on a Prisma-level).
+At a Prisma level, a relation is made up of:
+
+- Two [_relation fields_](#relation-fields): `author` and `posts`. Relation fields are used to define connections between models at the Prisma level and **do not exist in the database**.
+- The scalar `authorId` field. This field **does** exist in the database and represents the foreign key that connect a `Post` record to a `User` record.
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:29]: The second item here makes me stumble: 
Although it is true that the relation fields to not really exist, but the relation scalar field does, the "foreign key" part (which is a separate thing for me in SQL) is _not_ represented in the scalar field. This is indeed in the main relation field (which includes its target).

Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:29]: scalar => relation scalar to be exact?

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx [hunk 2]
```diff
@@ -888,58 +895,89 @@ Note that self-relations always require the `@relation` attribute.
...
+- "a user can have one or zero successors" (for example, Mary is Sarah's successor as blog owner)
 
-Note that one-to-one self-relations can not be made required on both sides, one field has to be optional.
+> **Note**: One-to-one self-relations cannot be made required on both sides. One or both sides must be optional, otherwise it becomes impossible to create the first `User` record.
 
-Self-relations always require the `@relation` attribute on both sides of the relation. On both ends, you need provide the same relation _name_.
+To create a one-to-one self-relation:
 
-The `husband` field on the other hand is the [annotated relation field](#annotated-relation-fields-and-relation-scalar-fields) (although here the term is ambiguous because both relation fields are annotated with the `@relation` attribute, however here this refers too the field that also has the `fields` and `references` arguments).
+- Both sides of the relation must define a `@relation` attribute that share the same name - in this case, **BlogOwnerHistory**.
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:919]: share the same name => share the same relation name

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx [hunk 3]
```diff
@@ -888,58 +895,89 @@ Note that self-relations always require the `@relation` attribute.
...
+- One relation field must be a [fully annotated](#annotated-relation-fields-and-relation-scalar-fields). In this example, the `successor` field defines both the `field` and `references` arguments.
+- One relation field must be backed by a foreign key. The `successor` field is backed by the `successorId` foreign key, which references a value in the `id` field.
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:921]: the bit about the database (old line 913) is important here in my opinion, as this sentence otherwise mixes terminology of Prisma and database

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx [hunk 4]
```diff
@@ -888,58 +895,89 @@ Note that self-relations always require the `@relation` attribute.
...
 
-The annotated field is backed by a foreign key in the underlying database. Here, `id` is the [relation scalar](#annotated-relation-fields-and-relation-scalar-fields) that's referenced by `husband`.
+> **Note**: One-to-one self relations require two sides even if both sides are equal in the relationship. For example, to model a 'best friends' relation, you would need to create two relation fields: `bestfriend1` and a `bestfriend2`.
 
-You can again determine which side of the relation should store the foreign key in the underlying database:
+Either side of the relation can be backed by a foreign key. In the following example, `successor` is backed by `successorId`:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:925]: https://github.com/prisma/docs/pull/375/files#r430071733

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx [hunk 5]
```diff
@@ -888,58 +895,89 @@ Note that self-relations always require the `@relation` attribute.
...
+  id             Int     @default(autoincrement()) @id
+  name           String?
+| successorId Int?
+| successor   User?   @relation("BlogOwnerHistory", fields: [successorId], references: [id])
+  predecessor User?   @relation("BlogOwnerHistory")
 }
 ```
 
-```prisma
+In the following example, `predecessor` is backed by `predecessorId`:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:937]: https://github.com/prisma/docs/pull/375/files#r430071733

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 1 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:29]: Hmmmm..I'm not sure I follow you. `authorId` is the foreign key in the database. Is the word 'scalar' in the sentence 'scalar `authorId` field' the part that doesn't work?

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 3 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:921]: "One relation field must be backed by a foreign key **in the underlying database**"?

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 4 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:925]: "Either side of the relation can be backed **by a foreign key in the underlying database**"?

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 1 above)
Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:29]: Let me rephrase:
1. `authorId` is a column in the database, that also has some FK stuff on it.
2. In Prisma schema we have a `authorId` field, but the FK stuff in represented in the relation field with `@relation`.

I am not sure line 29 represents that correctly by saying "represents the foreign key".

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 3 above)
Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:921]: Much better, yes.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx (hunk 4 above)
Comment (janpio) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/06-relations.mdx:925]: Much better, yes.
